batch/batch/driver/canceller.py

```diff
@@ -0,0 +1,171 @@
+import logging
+import asyncio
+
+from hailtop.utils import (WaitableSharedPool, retry_long_running, run_if_changed,
+                           AsyncWorkerPool)
+from hailtop import aiotools
+from gear import Database
+
+from ..batch import unschedule_job, mark_job_complete
+
+log = logging.getLogger('canceller')
+
+
+class Canceller:
+    def __init__(self, app):
+        self.app = app
+        self.cancel_ready_state_changed: asyncio.Event = app['cancel_ready_state_changed']
+        self.cancel_running_state_changed: asyncio.Event = app['cancel_running_state_changed']
+        self.db: Database = app['db']
+        self.async_worker_pool = AsyncWorkerPool(parallelism=100, queue_size=100)
```

Comment:
This means we allow as many as 200 parallel threads of control because there are two distinct pools each with 100 workers. I don't know how many parallel threads is OK, but this is definitely a change from the old code. Seems prudent to keep it the same by sharing a worker pool between canceller and scheduler.